core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -205,7 +205,11 @@ abstract class RawModule extends BaseModule {
       case (true, false) if left.probeInfo.get.writable => ProbeDefine(si, left.lref, RWProbeExpr(Node(right)))
       case (true, false)                                => ProbeDefine(si, left.lref, ProbeExpr(Node(right)))
       case (false, true)                                => Connect(si, left.lref, ProbeRead(Node(right)))
-      case (false, false)                               => Connect(si, left.lref, Node(right))
+      case (false, false) =>
+        (left, right) match {
+          case (_: Property[_], _: Property[_]) => PropAssign(si, left.lref, Node(right))
+          case (_, _) => Connect(si, left.lref, Node(right))
```

Comment:
What if `left: Property[_]` and `right: _`? this should be forbidden explicitly and throw it at elaboration time.

Reply:
True. Maybe I can delegate to MonoConnect which checks these things. Or just add the runtime check here.

Reply:
See my comment on the other thread--this would only happen if there was a bug in the implementation of BoringUtils, so I think it is reasonable to just deal with the two cases we care about.